broker-cli/cmd/flags/brokerurl.go

```diff
@@ -29,6 +38,22 @@ func ConstructBrokerURL(host, project, broker string) string {
 	return fmt.Sprintf("%s/v1beta1/projects/%s/brokers/%s", host, project, broker)
 }
 
+func (flags *BrokerURLConstructor) validateServer() error {
+	if !strings.HasPrefix(flags.Server, flags.Host) {
+		return fmt.Errorf("broker server URL %q should always begin with service end point %q", flags.Server, flags.Host)
+	}
+
+	parts := strings.Split(flags.Server, "/")
+	partsLen := len(parts)
+
+	if parts[partsLen-2] != brokersKey || parts[partsLen-4] != projectsKey || parts[partsLen-5] != betaKey {
```

Comment:
can this crash on some server flag inputs? (i.e. len(parts) <5)?

Reply:
it can. instead of just checking if partsLen < 5 and quitting , I am checkign for the correct partsLen, which should actually  be = 8